apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -164,12 +168,17 @@ defmodule Aecore.Chain.Worker do
       "Added block ##{new_block.header.height} with hash #{Base.encode16(new_block_hash)}, total tokens: #{inspect(total_tokens)}"
     end)
 
-    ## Store new block to disk
-    Persistence.write_block_by_hash(new_block)
     state_update1 = %{state | blocks_map: updated_blocks_map,
                               chain_states: updated_chain_states,
                               txs_index: new_txs_index}
     if top_height < new_block.header.height do
+      ## Store to RocksDB latest chain_state, latest block and latest block info
+      Persistence.batch_write(%{:chain_state => new_chain_state,
+                                :block => %{new_block_hash => new_block},
+                                :latest_block_info => %{"top_hash" => new_block_hash,
+                                                        "top_height" => new_block.header.height}})
+
+
```

Comment:
You should also write blocks to disk when `top_height >= new_block.header.height` (in the "else"), because this means we are just adding some fork and it might be longer, but the current block is "lower" then our top.

Reply:
fixed

Reply:
@thepiwo @cytadela8 is there anything else you want to add, that I'm missing ? 
@nedelcho-delchev-tues @d-velev can you review as well ? Thanks.